jerry-core/jcontext/jcontext.h

```diff
@@ -187,6 +190,36 @@ extern jerry_hash_table_t jerry_global_hash_table;
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#else /* JERRY_ENABLE_EXTERNAL_CONTEXT */
+
+#define JERRY_CONTEXT(field) (((jerry_context_t *) jerry_port_get_current_instance()->context_p)->field)
+
+#ifndef JERRY_SYSTEM_ALLOCATOR
+
+#define JERRY_HEAP_CONTEXT(field) (((jmem_heap_t *) jerry_port_get_current_instance ()->lcache_p)->field)
+
+#undef JMEM_HEAP_SIZE
+#define JMEM_HEAP_SIZE (jerry_port_get_current_instance ()->heap_size)
+
+#undef JMEM_HEAP_AREA_SIZE
+#define JMEM_HEAP_AREA_SIZE (jerry_port_get_current_instance ()->heap_size - JMEM_ALIGNMENT)
+
```

Comment:
Hmm.. yeah..

I guess another advantage of the "one big slab of mem" approach is that these getters would be part of jerry-core and the user would only have to provide a getter for that one big buffer.